src/api/spec/controllers/webui/user_controller_spec.rb

```diff
@@ -340,20 +340,26 @@
   end
 
   describe 'GET #icon' do
-    let(:user) { create(:confirmed_user, login: 'iconic') }
+    context 'with an inexistent user' do
+      before do
+        get :icon, params: { user: 'inexistent_user' }
+      end
 
-    it 'loads big icon without param' do
-      get :icon, params: { user: user.login }
-      image = ChunkyPNG::Image.from_blob(response.body)
-      expect(image.height).to be 80
-      expect(image.width).to be 80
+      it { expect(response.location).to match(/.*\/assets\/default_face-.*\.png$/) }
+      it { expect(response).to have_http_status(302) }
     end
 
-    it 'loads small icon with param' do
-      get :icon, params: { user: user.login, size: 20 }
-      image = ChunkyPNG::Image.from_blob(response.body)
-      expect(image.height).to be 20
-      expect(image.width).to be 20
+    context 'with a user not having an icon' do
+      let(:user) { instance_double('User', login: 'test_user') }
+
+      before do
+        get :icon, params: { user: user.login, size: 20 }
+
+        allow(user).to receive(:gravatar_image).with(20).and_return(:none)
+      end
+
+      it { expect(response.location).to match(/.*\/assets\/default_face-.*\.png$/) }
```

Comment:
what you don't test at all anymore is that the route does something proper with the size param. It's just a matter of time that a future refactoring will make sure we load wallpaper sized icons into an <img width=20>